code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,684 @@
-		return
-
-	if(attached.species.flags & NO_BLOOD)
+	if(istype(W, /obj/item/clothing/mask/breath))
+		if(is_type_in_list(W, mask_blacklist))
+			to_chat(usr, "\The [W] is incompatible with \the [src].")
+			return
+		if(breath_mask)
+			to_chat(usr, "There is already a mask installed.")
+			return
+		usr.drop_from_inventory(W, src)
+		breath_mask = W
+		usr.visible_message(SPAN_NOTICE("[usr] places \the [W] in \the [src]."), SPAN_NOTICE("You place \the [W] in \the [src]."))
+		update_icon()
 		return
-
-	// Give blood
-	if(mode)
-		if(beaker.reagents.total_volume > 0)
-			beaker.reagents.trans_to_mob(attached, transfer_amount, CHEM_BLOOD)
-			update_icon()
-
-		if(toggle_stop) // Automatically detaches if the blood volume is at 100%
-			if(beaker.reagents.has_reagent(/decl/reagent/blood) && attached.get_blood_volume() >= 100)
-				visible_message("[icon2html(src, viewers(get_turf(src)))] \The <b>[src]</b> flashes a warning light, disengaging from [attached] automatically!")
-				playsound(src, 'sound/machines/buzz-two.ogg', 50)
-				src.attached = null
-				src.update_icon()
-				blood_message_sent = FALSE
-				update_icon()
+	if(istype(W, /obj/item/tank))
+		if(is_type_in_list(W, tank_blacklist))
+			to_chat(usr, "\The [W] is incompatible with \the [src].")
+			return
+		if(tank)
+			to_chat(usr, "There is already a tank installed!")
+			return
+		if(istype(W, /obj/item/tank/phoron))
+			if(tipped)
+				to_chat(usr, "You're not sure how to place \the [W] in the fallen [src].")
 				return
-
-	// Take blood
-	else
-		var/amount = REAGENTS_FREE_SPACE(beaker.reagents)
-		amount = min(amount, transfer_amount)
-		// If the beaker is full, ping
-		if(amount == 0)
-			if(prob(5))
-				visible_message("[src] pings.")
+		usr.drop_from_inventory(W, src)
+		tank = W
+		usr.visible_message(SPAN_NOTICE("[usr] places \the [W] in \the [src]."), SPAN_NOTICE("You place \the [W] in \the [src]."))
+		update_icon()
+		return
+	if(W.isscrewdriver())
+		if(!tank)
+			to_chat(usr, "There isn't a tank installed for you to secure!")
 			return
-
-		if(attached.get_blood_volume() < 90 && !blood_message_sent)
-			visible_message("[icon2html(src, viewers(get_turf(src)))] \The <b>[src]</b> flashes a warning light!")
-			playsound(src, 'sound/machines/buzz-two.ogg', 50)
-			blood_message_sent = TRUE
-
-		if(attached.take_blood(beaker,amount))
-			update_icon()
+		if(tank_type == "phoron")
+			to_chat(usr, "You can't properly secure this type of tank to \the [src]!")
+			return
+		usr.visible_message(
+			SPAN_NOTICE("[usr] [is_loose ? "tightens" : "loosens"] the screws on [src]."),
+			SPAN_NOTICE("You [is_loose ? "tighten" : "loosen"] the screws on [src], [is_loose ? "securing \the [tank]" : "allowing \the [tank] to be removed"]."))
+		playsound(src.loc, "sound/items/[pick("Screwdriver", "Screwdriver2")].ogg", 50, 1)
+		is_loose = !is_loose
+		return
+	return ..()
 
 /obj/machinery/iv_drip/attack_hand(mob/user as mob)
-	if (isAI(user))
+	if(isAI(user))
```

Comment:
I think this check should instead be done with a `/proc/attack_ai()` or `/proc/ai_can_interact()` override / something.